flake.nix

```diff
@@ -0,0 +1,48 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/master";
+    flake-utils.url = "github:numtide/flake-utils";
+    gomod2nix = {
+      url = "github:nix-community/gomod2nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+      inputs.utils.follows = "flake-utils";
+    };
+  };
+
+  outputs = { self, nixpkgs, gomod2nix, flake-utils }:
+    {
+      overlays.default = pkgs: _: {
+        simd = pkgs.callPackage ./simapp { rev = self.shortRev or "dev"; };
+      };
+    } //
+    (flake-utils.lib.eachDefaultSystem
+      (system:
+        let
+          mkApp = drv: {
+            type = "app";
+            program = "${drv}/bin/${drv.meta.mainProgram}";
+          };
+          pkgs = import nixpkgs {
+            inherit system;
+            config = { };
+            overlays = [
+              gomod2nix.overlays.default
+              self.overlays.default
+            ];
+          };
+        in
+        rec {
+          packages = { inherit (pkgs) simd; };
+          apps = { simd = mkApp pkgs.simd; };
+          defaultPackage = packages.simd;
+          defaultApp = apps.simd;
+          devShell = with pkgs; mkShell {
+            buildInputs = [
+              go_1_20
```

Comment:
Can we add a useless comment like `Use Go 1.20 version`, that helps when we search / replace the Go version :D

Reply:
sure, done